nixos/modules/services/networking/6tunnel.nix

```diff
@@ -0,0 +1,111 @@
+{ lib, config, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services._6tunnel;
+
+in {
+  options.services._6tunnel = {
+
+    tunnels = mkOption {
+      default = [];
+      description = "Which tunnels to create.";
+      type = with types; listOf (submodule {
+        options = {
+          port = mkOption {
+            description = "Local port to open.";
+            type = types.port;
+          };
+          host = mkOption {
+            description = "Remote host to tunnel to.";
+            type = types.str;
+          };
+          remotePort = mkOption {
+            default = null;
+            description = "Port on remote host. If null (default), it uses the the local port.";
+            type = with types; nullOr port;
+          };
+          restart = mkOption {
+            default = null;
+            description = ''
+              Whether to restart the tunnel periodically, which is useful for non-static IP addresses.
+              The format is described in SYSTEMD.TIME(7).
+
+              If null (default), the tunnel won't restart periodically.
+            '';
+            type = with types; nullOr str;
+          };
+        };
+      });
+      example = [
+        { port = "8080"; host = "192.168.0.2"; remotePort = 80; }
+        { port = "443"; host = "www.example.com"; }
+        { port = "8053"; host = "dynamicip.example.com"; restart = "1h"; }
+      ];
+    };
+
+    openFirewall = mkOption {
+      default = true;
+      description = "Whether to open the needed firewall ports.";
+      type = types.bool;
+    };
```

Comment:
Did you do this for the whole service and not per tunnel on purpose?

Reply:
No, I just didn't think it would be useful per tunnel, I'll change it.

Reply:
That's why I'm asking: I've never used 6tunnel, so you would know what's useful.

Reply:
If I think about it: you would only block the firewall if you want to open ports manually (e.g. restrict them to certain IPs) and then it makes sense to just have one option globally instead of per tunnel. This service doesn't make any sense without opening ports, so I'll leave it as a global option.